flake8_quotes/__init__.py

```diff
@@ -119,32 +107,24 @@ def parse_options(cls, options):
             #   -> {good_single: ', good_multiline: ''', bad_single: ", bad_multiline: """}
             cls.config.update(cls.MULTILINE_QUOTES[options.multiline_quotes])
 
-    def get_file_contents(self):
-        if self.filename in ('stdin', '-', None):
-            return stdin_get_value().splitlines(True)
-        else:
-            return readlines(self.filename)
-
     def run(self):
-        file_contents = self.get_file_contents()
-
-        noqa_line_numbers = self.get_noqa_lines(file_contents)
-        errors = self.get_quotes_errors(file_contents)
-
-        for error in errors:
+        self._tokens = [Token(t)
+                        for t in tokenize.generate_tokens(
```

Comment:
**Required discussion:** After doing some research, it looks like `flake8` supports passing tokens instead of lines in exactly the same manner as this. Is there any reason we aren't accepting `tokens`/using `file_tokens` as an `__init__` parameter?

https://github.com/PyCQA/flake8/blob/3.2.1/src/flake8/processor.py#L96-L124

http://flake8.pycqa.org/en/latest/plugin-development/plugin-parameters.html#indicating-desired-data

This has been around for at least 2 years so we're happy to use it